spacy/language.py

```diff
@@ -1576,9 +1580,22 @@ def pipe(  # noqa: F811
             docs = (self._ensure_doc(text) for text in texts)
             for pipe in pipes:
                 docs = pipe(docs)
+
         for doc in docs:
             yield doc
 
+    def _prepare_doc_for_ipc(
+        self, texts: Iterable[Union[str, Doc]]
+    ) -> Iterable[Tuple[Union[str, bytes], _AnyContext]]:
+        # Serialize Doc inputs to bytes to avoid incurring pickling
+        # overhead when they are passed to child processes. Also yield
+        # any context objects they might have separately (as they are not serialized).
+        for doc_like in texts:
+            if isinstance(doc_like, Doc):
+                yield (doc_like.to_bytes(), cast(_AnyContext, doc_like._context))
+            else:
+                yield (doc_like, cast(_AnyContext, None))
+
```

Comment:
I think this could be defined within `_multiprocessing_pipe` and then it could have a simpler name like `prepare_doc`?